From 55d04c40d9a4f51ff7961226658559fb01a97ee3 Mon Sep 17 00:00:00 2001
From: Anakin Childerhose <anakin@childerhose.ca>
Date: Sat, 7 Dec 2024 09:53:31 -0500
Subject: [PATCH] remove theme icons since I don't use them
Upstream-Status: Pending

---
 CMakeLists.txt      | 42 ------------------------------------------
 awesomeConfig.cmake |  2 --
 2 files changed, 44 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c05a7fbc..c43528d0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -318,48 +318,6 @@ if(GENERATE_DOC)
 endif()
 # }}}
 
-# {{{ Theme icons
-file(GLOB_RECURSE icon_sources RELATIVE ${SOURCE_DIR} ${SOURCE_DIR}/themes/*.png)
-
-foreach(icon ${icon_sources})
-    # Copy all icons to the build dir to simplify the following code.
-    # Source paths are interpreted relative to ${SOURCE_DIR}, target paths
-    # relative to ${BUILD_DIR}.
-    get_filename_component(icon_path ${icon} PATH)
-    get_filename_component(icon_name ${icon} NAME)
-    file(COPY ${icon} DESTINATION ${icon_path})
-    set(ALL_ICONS ${ALL_ICONS} "${icon_path}/${icon_name}")
-endforeach()
-
-macro(a_icon_convert match replacement input)
-    string(REPLACE ${match} ${replacement} output ${input})
-
-    if(NOT ${input} STREQUAL ${output} AND NOT ";${ALL_ICONS};" MATCHES ";${output};")
-        set(ALL_ICONS ${ALL_ICONS} ${output})
-
-        add_custom_command(
-            COMMAND ${CONVERT_EXECUTABLE} ${input} -strip ${ARGN} ${output}
-            OUTPUT  ${output}
-            DEPENDS ${input}
-            VERBATIM)
-    endif()
-endmacro()
-
-foreach(icon ${ALL_ICONS})
-    # Make unfocused icons translucent
-    a_icon_convert("_focus" "_normal" ${icon}
-        -colorspace rgb -gamma 0.6 -channel A -evaluate Multiply 0.4)
-endforeach()
-
-foreach(icon ${ALL_ICONS})
-    # Make inactive icons grayscale
-    a_icon_convert("_active" "_inactive" ${icon}
-        -colorspace Gray)
-endforeach()
-
-add_custom_target(generated_icons ALL DEPENDS ${ALL_ICONS})
-# }}}
-
 # {{{ Dist tarball
 if(BUILD_FROM_GIT)
     add_custom_target(dist
diff --git a/awesomeConfig.cmake b/awesomeConfig.cmake
index 592c2b25..93ead990 100644
--- a/awesomeConfig.cmake
+++ b/awesomeConfig.cmake
@@ -59,8 +59,6 @@ if(GENERATE_DOC)
 else()
     message(STATUS "Not generating API documentation.")
 endif()
-# theme graphics
-a_find_program(CONVERT_EXECUTABLE convert TRUE)
 # pkg-config
 include(FindPkgConfig)
 # lua
