From 37d70288d263d0cd598b098ece0636f60b6e0a7c Mon Sep 17 00:00:00 2001
From: Anakin Childerhose <anakin@childerhose.ca>
Date: Mon, 23 Dec 2024 23:11:19 -0500
Subject: [PATCH] build: allow for cross compilation
Upstream-Status: Pending

---
 makefile | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/makefile b/makefile
index 41a2928..53e5c72 100644
--- a/makefile
+++ b/makefile
@@ -24,9 +24,10 @@ CWARNS = -Wall -Wextra -pedantic \
 	-Wc++-compat \
 # -Wunreachable-code \
 
-
-CFLAGS = $(CWARNS) $(COPT) -std=c99 -I$(LUADIR) -fPIC
-CC = gcc
+CFLAGS ?= $(CWARNS) $(COPT) -std=c99 -I$(LUADIR) -fPIC
+CFLAGS += -fPIC
+LDFLAGS += -shared -fPIC
+CC ?= gcc
 
 FILES = lpvm.o lpcap.o lptree.o lpcode.o lpprint.o lpcset.o
 
@@ -39,7 +40,7 @@ macosx:
 	$(MAKE) lpeg.so "DLLFLAGS = -bundle -undefined dynamic_lookup"
 
 lpeg.so: $(FILES)
-	env $(CC) $(DLLFLAGS) $(FILES) -o lpeg.so
+	env $(CC) $(LDFLAGS) $(FILES) -o lpeg.so
 
 $(FILES): makefile
 
